src/catkin_pkg/condition.py

```diff
@@ -57,8 +57,27 @@ def _evaluate(parse_results, context):
         # return literal value
         return parse_results
 
+    assert len(parse_results) >= 3
```

Comment:
This is ensuring the same.:blush:

Reply:
I thought that the removed assertion was adding to redundancy in the code, so removed it.

Reply:
No, this line does **not** ensure that. It only ensures >= 3 but not that it must be odd.

Reply:
Done

Reply:
@dirk-thomas , can you please review it?